src/main/java/com/mongodb/kafka/connect/source/schema/BsonDocumentToSchema.java

```diff
@@ -50,12 +52,11 @@ private static Schema inferDocumentSchema(final String fieldPath, final BsonDocu
   private static Schema inferArraySchema(final String fieldPath, final BsonArray bsonArray) {
     Schema combinedSchema = SENTINEL_STRING_TYPE;
     for (final BsonValue v : bsonArray) {
-      combinedSchema = combineArrayValueSchema(combinedSchema, inferSchema(fieldPath, v));
+      combinedSchema = combinedSchema(combinedSchema, inferSchema(fieldPath, v));
       if (combinedSchema == INCOMPATIBLE_SCHEMA_TYPE) {
         break;
       }
     }
-    combinedSchema = isSentinel(combinedSchema) ? Schema.OPTIONAL_STRING_SCHEMA : combinedSchema;
```

Comment:
Had to remove this as this loses the sentinel.  

Note: The sentinel is equivalent to Schema.OPTIONAL_STRING_SCHEMA but not the same instance as it and that is what we check against.